docs: Reorganize documentation on the webpage

kelbrown20 opened the pull request:
# What does this PR do?
 Reorganizes the Llama stack webpage into more concise index pages, introduce more of a workflow, and reduce repetition of content.

New nav structure so far based on #2637 

Further discussions in https://github.com/meta-llama/llama-stack/discussions/2585

**Preview:**
![Screenshot 2025-07-09 at 2 31 53 PM](https://github.com/user-attachments/assets/4c1f3845-b328-4f12-9f20-3f09375007af)

You can also build a full local preview locally 

 **Feedback**
Looking for feedback on page titles and general feedback on the new structure

**Follow up documentation**
I plan on reducing some sections and standardizing some terminology in a follow up PR.
More discussions on that in https://github.com/meta-llama/llama-stack/discussions/2585

reluctantfuturist commented:
Thank you and welcome @kelbrown20 ! I added a sub-issue to track your work on documentation structure: https://github.com/meta-llama/llama-stack/issues/2653

nathan-weinberg commented on docs/source/concepts/architecture.md:
```diff
@@ -1,31 +1,41 @@
-# Why Llama Stack?
+## Llama Stack architecture  
+
+Llama stack allows you to build different layers of distributions to for your AI workloads using various SDKs and API providers.
```
```suggestion
Llama Stack allows you to build different layers of distributions for your AI workloads using various SDKs and API providers.
```

nathan-weinberg commented on docs/source/advanced_apis/index.md:
```diff
@@ -0,0 +1,37 @@
+# Advanced API Providers
```
Curious - why "Advanced API Providers" and not just "APIs" ?

kelbrown20 replied:
Yeah I agree, I think Advanced APIs is better

leseb reviewed (changes requested):
Looks goog overall.

leseb commented:
Will this close https://github.com/meta-llama/llama-stack/issues/2681 entirely?

kelbrown20 commented on docs/source/advanced_apis/post_training/inline_huggingface.md:
```diff
@@ -0,0 +1,36 @@
+{input}
+<|assistant|>
+{output} |  |
+| `model_specific_config` | `<class 'dict'>` | No | {'trust_remote_code': True, 'attn_implementation': 'sdpa'} |  |
```
Just fixed the format!

kelbrown20 commented on docs/source/deploying/openai.md:
```diff
@@ -1,14 +1,14 @@
-# OpenAI API Compatibility
+## OpenAI API Compatibility
```
Just moved that

kelbrown20 commented:
@leseb Just fixed what you mentioned!

kelbrown20 commented:
Yeah thats a great point @bbrowning!
I added the `sphynx-redirects` extension and added redirects for the files that I changed [here](https://github.com/meta-llama/llama-stack/pull/2651/files#diff-008dcb3426febd767787b1521f1fe33086313b927ea37eaab86df5fa88a51698R58). Lmk if there was anything I might have missed

I can also made another PR that updates Contributor guide to show this process, if that helps. There also might be a more. automated way of doing this that I can look into as well.

bbrowning commented:
Nice, thanks for the redirects! I pulled this locally and the redirects work well.

One note, I had to edit my `pyproject.toml` locally to add `"sphinx-reredirects",` in the `docs = [ ... ]` section where it lists the docs dependencies. And, afterwards, I had to run `uv run pre-commit run --all-files` to regenerate the uv.lock file.

We'll need to add that to this PR to actually get the redirects working once this merges.

kelbrown20 commented:
@bbrowning I fixed some of the warnings we discussed, but also made an issue that can tasks fixing the other warnings

bbrowning reviewed (approved):
Looks great! Thanks for all the quality of life fixes you tackled as part of this, in addition to updating the content itself 😄

raghotham commented:
Thanks for the reorg. Do you mind taking a look at all of these warnings and fix them?
